Mathlib/LinearAlgebra/Matrix/ToLin.lean

```diff
@@ -984,3 +984,85 @@ lemma end_apply_apply (ij : ι × ι) (k : ι) : (b.end ij) (b k) = if ij.2 = k
   linearMap_apply_apply b b ij k
 
 end Basis
+
+section
+
+variable (ι : Type*) [Fintype ι] [DecidableEq ι]
+variable (R : Type*) [Semiring R]
+variable (M : Type*) [AddCommMonoid M] [Module R M]
+
+/--
+Let `M` be an `R`-module. Every `R`-linear map `Mⁿ → Mⁿ` corresponds to a `n×n`-matrix whose entries
+are `R`-linear maps `M → M`. In another word, we have`End(Mⁿ) ≅ Matₙₓₙ(End(M))` defined by:
+`(f : Mⁿ → Mⁿ) ↦ (x ↦ f (0, ...,x at j-th position, ..., 0) i)ᵢⱼ` and
+`m : Matₙₓₙ(End(M)) ↦ (v ↦ ∑ⱼ mᵢⱼ(vⱼ))`.
+
+See also `LinearMap.toMatrix'`
+-/
+@[simps]
+def endVecEquivMatrixEnd :
+    Module.End R (ι → M) ≃ Matrix ι ι (Module.End R M) where
```

Comment:
Why not inlining this into the RingEquiv?

Reply:
Thanks